
* modules/loggers/mod_log_debug.c: Mark private globals as static.
http://svn.apache.org/viewvc?view=revision&revision=1326980

* modules/filters/sed1.c: Mark private globals as static const.
  (command): Change p3 pointer to const. 
http://svn.apache.org/viewvc?view=revision&revision=1326984

* modules/filters/config.m4: Prevent libsed internals from polluting
  the global symbol namespace.
http://svn.apache.org/viewvc?view=revision&revision=1326991

Upstream-Status: Backport

--- httpd-2.4.2/modules/loggers/mod_log_debug.c
+++ httpd-2.4.2/modules/loggers/mod_log_debug.c
@@ -35,8 +35,8 @@
     apr_array_header_t *entries;
 } log_debug_dirconf;
 
-const char *allhooks = "all";
-const char * const hooks[] = {
+static const char *allhooks = "all";
+static const char * const hooks[] = {
     "log_transaction",      /*  0 */
     "quick_handler",        /*  1 */
     "handler",              /*  2 */
--- httpd-2.4.2/modules/filters/sed1.c
+++ httpd-2.4.2/modules/filters/sed1.c
@@ -25,7 +25,7 @@
 #include "apr_strings.h"
 #include "regexp.h"
 
-char    *trans[040]  = {
+static const char *const trans[040]  = {
     "\\01",
     "\\02",
     "\\03",
@@ -58,7 +58,7 @@
     "\\36",
     "\\37"
 };
-char rub[] = {"\\177"};
+static const char rub[] = {"\\177"};
 
 extern int sed_step(char *p1, char *p2, int circf, step_vars_storage *vars);
 static int substitute(sed_eval_t *eval, sed_reptr_t *ipc,
@@ -692,7 +692,8 @@
                             step_vars_storage *step_vars)
 {
     int    i;
-    char   *p1, *p2, *p3;
+    char   *p1, *p2;
+    const char *p3;
     int length;
     char sz[32]; /* 32 bytes enough to store 64 bit integer in decimal */
     apr_status_t rv = APR_SUCCESS;
--- httpd-2.4.2/modules/filters/config.m4
+++ httpd-2.4.2/modules/filters/config.m4
@@ -16,7 +16,13 @@
 APACHE_MODULE(substitute, response content rewrite-like filtering, , , most)
 
 sed_obj="mod_sed.lo sed0.lo sed1.lo regexp.lo"
-APACHE_MODULE(sed, filter request and/or response bodies through sed, $sed_obj, , most)
+APACHE_MODULE(sed, filter request and/or response bodies through sed, $sed_obj, , most, [
+    if test "x$enable_sed" = "xshared"; then
+        # The only symbol which needs to be exported is the module
+        # structure, so ask libtool to hide libsed internals:
+        APR_ADDTO(MOD_SED_LDADD, [-export-symbols-regex sed_module])
+    fi
+])
 
 if test "$ac_cv_ebcdic" = "yes"; then
 # mod_charset_lite can be very useful on an ebcdic system,
